src/ORM/Hierarchy/MarkedSet.php

```diff
@@ -459,21 +459,41 @@ public function markPartialTree()
 
         // Build markedNodes for this subtree until we reach the threshold
         // foreach can't handle an ever-growing $nodes list
-        while (list(, $node) = each($this->markedNodes)) {
-            $children = $this->markChildren($node);
-            if ($nodeCountThreshold && sizeof($this->markedNodes) > $nodeCountThreshold) {
-                // Undo marking children as opened since they're lazy loaded
-                /** @var DataObject|Hierarchy $child */
-                foreach ($children as $child) {
-                    $this->markClosed($child);
+        $done = [];
+        while ($nodes = $this->iterateNewItems($this->markedNodes, $done)) {
+            foreach ($nodes as $node) {
+                $children = $this->markChildren($node);
+                if ($nodeCountThreshold && sizeof($this->markedNodes) > $nodeCountThreshold) {
```

Comment:
Would this `sizeof()` cause warnings as well?

Reply:
No because $this->markedNodes isn't a callable.